llvm/lib/CAS/Utils.cpp

```diff
@@ -84,3 +84,34 @@ Error cas::walkFileTreeRecursively(
 
   return Error::success();
 }
+
+Error cas::walkObjectsRecursively(
```

Comment:
I will add a comment in the header for what this function does. This is a Preorder traversal of the object DAG.